python/extractor/semmle/path_filters.py

```diff
@@ -41,19 +41,27 @@ def glob_part_to_regex(glob, add_sep):
 
 def glob_to_regex(glob, prefix=""):
     '''Convert entire glob to a compiled regex'''
+    # When the glob ends in `/`, we need to remember this so that we don't accidentally add an
+    # extra separator to the final regex.
+    end_sep = "" if glob.endswith("/") else SEP
     glob = glob.strip().strip("/")
     parts = glob.split("/")
     #Trailing '**' is redundant, so strip it off.
     if parts[-1] == "**":
         parts = parts[:-1]
         if not parts:
             return ".*"
+    # The `glob.strip("/")` call above will have removed all trailing slashes, but if there was at
+    # least one trailing slash, we want there to be an extra part, so we add it explicitly here in
+    # that case, using the emptyness of `end_sep` as a proxy.
+    if end_sep == "":
+        parts += [""]
```

Comment:
As I understand it, this is equivalent to:
```python
glob = glob.strip().lstrip("/").rstrip_only_repeated("/")
end_sep = "" if glob.endswith("/") else SEP
parts = glob.split("/")
#Trailing '**' is redundant, so strip it off.
    if parts[-1] == "**":
        parts = parts[:-1]
        if not parts:
            return ".*"
```
which might be simpler, if `rstrip_only_repeated` was something that existed :-)

I will not insist on rewriting this, though.

Reply:
Indeed. In hindsight, perhaps I should have done
```python
end_sep = ...
glob = glob.strip().strip("/")
if end_sep == "":
    glob += "/"
```
Or perhaps an even better idea would be to first transform all runs of `/` into single occurrences, and then only do `lstrip`.

I think I'll leave it as-is for now. At some point we can hopefully just get rid of this code altogether.